src/primitives/transaction.cpp

```diff
@@ -57,8 +57,10 @@ JSDescription JSDescription::Randomized(
     // Randomize the order of the inputs and outputs
     inputMap = {0, 1};
     outputMap = {0, 1};
-    MappedShuffle(inputs.begin(), inputMap.begin(), ZC_NUM_JS_INPUTS, gen);
-    MappedShuffle(outputs.begin(), outputMap.begin(), ZC_NUM_JS_OUTPUTS, gen);
+    if (gen) {
```

Comment:
(The reason being that with this conditional section, it is no longer obvious to the caller that if they accidentally pass in an empty function, `JSDescription::Randomized` no longer does what it says on the box. With an explicit identity function, it is obvious.)